### PR TITLE
fix(plot): add error handling for to_dataframe() if size 0

### PR DESCRIPTION
**Pull request by atsushi421**
@hsgwa 
This PR adds error handling if the size of DataFrame obtained by the to_dataframe() method is 0.
https://emb4hq.slack.com/archives/C017YM4AA06/p1662622513491059?thread_ts=1659948870.138049&cid=C017YM4AA06

To Reproduce: https://drive.google.com/drive/folders/1qtM778Ob7Chv-XAUuaOfUucKosDMCTyd?usp=sharing

### CODE REVIEW

File: src/caret_analyze/plot/bokeh/callback_info.py [hunk 1]
```diff
@@ -56,6 +58,10 @@ def _create_latency_df(
         callback: CallbackBase
     ) -> pd.DataFrame:
         df = callback.to_dataframe()
+        if len(df) == 0:
+            logger.warning('Since no timestamp is recorded, '
+                           'the latency cannot be calculated. '
+                           f'callback_name: {callback.callback_name}')
```
Comment (hsgwa) [inline src/caret_analyze/plot/bokeh/callback_info.py:64]: This implementation is redundant.
It's preferred to move error cases to "_to_dataframe_core".

```
for cb in self._callbacks:
    try:
        latency_df = self._create_latency_df(xaxis_type, cb)
        concat_latency_df = pd.concat([concat_latency_df, latency_df],
                                      axis=1)
    exception IndexError as e:
        df = cb.to_dataframe() 
        if (len(df) == 0) :
            logger.warning()
```

**Review by hsgwa — CHANGES_REQUESTED** [review]

File: src/caret_analyze/plot/bokeh/callback_info.py (hunk 1 above)
Comment (atsushi421) [reply src/caret_analyze/plot/bokeh/callback_info.py]: @hsgwa 
I have fixed by https://github.com/tier4/CARET_analyze/pull/130/commits/3063b537f2c6a302c8da0fbcdd4d3e2a41558bea and https://github.com/tier4/CARET_analyze/pull/130/commits/a9a63f5caa4828692d6cb8b0ae59f51070d850cf.
Could you please review it again?

Comment (hsgwa) [reply src/caret_analyze/plot/bokeh/callback_info.py]: Sorry, that wasn't good advice.
To guarantee  existence of source_df column, following implementation seems better.

```
for cb in self._callbacks:
    try:
        latency_df = self._create_latency_df(xaxis_type, cb)
        concat_latency_df = pd.concat([concat_latency_df, latency_df],
                                      axis=1)
    except IndexError as e:
        pass # or logger.warning
   finally:
        if column in source_df:
            pass # insert empty df
        if (len(cb.to_dataframe() ) == 0) :
            logger.warning()
```

**Review by hsgwa — CHANGES_REQUESTED** [review]
@atsushi421 

This patch raises errors in my environment.
Could you confirm these results?
https://drive.google.com/drive/folders/1khOVV6U3awJX2Ux6n5jLxTDUHjOKwlTf?usp=sharing

**Comment by atsushi421** [comment]
> @atsushi421
> 
> This patch raises errors in my environment. Could you confirm these results? https://drive.google.com/drive/folders/1khOVV6U3awJX2Ux6n5jLxTDUHjOKwlTf?usp=sharing

@hsgwa 
This is because Publisher.callback_name is None.
I have corrected this by https://github.com/tier4/CARET_analyze/pull/133

File: src/caret_analyze/plot/bokeh/callback_info.py (hunk 1 above)
Comment (atsushi421) [reply src/caret_analyze/plot/bokeh/callback_info.py]: @hsgwa 
Fixed in https://github.com/tier4/CARET_analyze/pull/130/commits/a04cae08875770c39a2f5fd7642b943b07b55740 :bow:

**Comment by hsgwa** [comment]
@atsushi421 

> This is because Publisher.callback_name is None.
OK. I'll review and merge this PR after #133 and #134

File: src/caret_analyze/plot/bokeh/pub_sub_info.py [hunk 2]
```diff
@@ -32,10 +35,25 @@ def __init__(
...
-                                         axis=1)
+        for i, pub_sub in enumerate(self._pub_subs, 1):
+            try:
+                period_df = self._create_period_df(xaxis_type, pub_sub)
+                concat_period_df = pd.concat([concat_period_df, period_df],
+                                             axis=1)
+            except IndexError:
+                pass
+            finally:
+                if i*2 != len(concat_period_df.columns):
```
Comment (hsgwa) [inline src/caret_analyze/plot/bokeh/pub_sub_info.py:46]: 関数にしたり、一旦変数に入れると意図が伝わりやすいです

```
has_new_column = i*2 == len(concat_period_df.columns)
if not has_new_column:
    pass
```

参考
- https://qiita.com/shimataro999/items/ef6cd838d56f1fe87015#%E6%9D%A1%E4%BB%B6%E5%BC%8F%E3%81%AF%E9%96%A2%E6%95%B0%E5%8C%96%E3%81%97%E3%81%A6%E7%9B%AE%E7%9A%84%E3%81%8C%E3%82%8F%E3%81%8B%E3%82%8B%E3%82%88%E3%81%86%E3%81%AB%E3%81%99%E3%82%8B

- https://murashun.jp/article/programming/naming-conventions.html

**Comment by atsushi421** [comment]
@hsgwa 
Conflicts were resolved.
Could you review it? :bow:

**Review by hsgwa — APPROVED** [review]
LGTM!
All bugs seem fixed.

**Pull request merged**